### PR TITLE
Unify nightly naming across binaries and docker images

### PR DESCRIPTION
**Pull request by techknowlogick**
Proposal found here: https://github.com/go-gitea/gitea/issues/23654

TODO: make non-breaking (can we publish docker image using dev and nightly prefix? at same time). if anyone has advice please comment :) 

If this PR is merged, then I can add redirects to the downloads site.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,6 +97,11 @@ else
 	endif
 endif
 
+# if version = "main" then update version to "nightly"
+ifeq ($(VERSION),main)
+	VERSION := nightly
```
Comment (wxiaoguang) [inline Makefile]: `nightly` or `main-nightly`?

1. `nightly`
   * If we tell users "please use nightly" build and they see "night" and "1.19-nightly", they might be confused.
   * If we tell users "please use main nightly" build they won't find such name.

2. `main-nightly`:
   * If we tell users "please use main-nightly" , then users will find "main-night" build.

**Comment by 6543** [comment]
> can we publish docker image using dev and nightly prefix? at same time

Well yes ... it just depend how complex our docker manifest build step is going to get

File: Makefile (hunk 1 above)
Comment (6543) [reply Makefile]: I trust the common sense ... and you an just tell "use the 1.19 nigly" if you dont think of main

**Comment by wxiaoguang** [comment]
I would propose to drop the `-dev` suffix. Because the `-dev` tags are not the latest , they are just the starting point of a branch:

* https://github.com/go-gitea/gitea/tree/v1.20.0-dev on Feb 22, 2023
* https://github.com/go-gitea/gitea/tree/v1.19.0-dev on Oct 26, 2022

I do not think it make sense to continue using the `-dev` suffix.

And I think almost all edge version users have the ability to fix there deployed image name.

File: Makefile (hunk 1 above)
Comment (wxiaoguang) [reply Makefile]: I agree with "common sense" makes sense, but many users might not know the "nightly" concept clearly. 
"main-nightly" doesn't violate common sense IMO 😄  

For naming style 1: If I tell a user to use "nightly", they might need to search "nightly", and then find 1.19-nightly, wrong .... A problem is that `1.19-nightly` might appear before `nightly` in the search result (order by string, eg: dockerhub search).

For naming style 2: I just need to tell the user to use "**main-nightly**", then the user goes to the download list, search & find "main-nightly", no ambiguity, use it, perfect.

That's my opinion and preference (`main-nightly`), ~~a little strong~~ but not blocker if most people think `nightly`+`1.19-nightly` is clearer and the word `nightly` wouldn't mislead users.

**Review by silverwind — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (techknowlogick) [reply Makefile]: I agree with this logic as well and have changed it to be `main-nightly`.

**Review by lunny — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**